src/utils.zig

```diff
@@ -49,10 +50,10 @@ pub fn readStructForeign(reader: io.StreamSource.Reader, comptime T: type) !T {
                 @field(result, entry.name) = try readStructForeign(reader, entry.field_type);
             },
             .Enum => {
-                @field(result, entry.name) = try reader.readEnum(entry.field_type, switch (native_endian) {
+                @field(result, entry.name) = reader.readEnum(entry.field_type, switch (native_endian) {
                     builtin.Endian.Little => builtin.Endian.Big,
                     builtin.Endian.Big => builtin.Endian.Little,
-                });
+                }) catch return error.InvalidData;
```

Comment:
Those methods should not be responsible to return error required by the Image interface